expertsystem/state/particle.py

```diff
@@ -542,6 +545,23 @@ def safe_set_spin_projections(
     ]
 
 
+def generate_outer_edge_permutations(
+    topology: StateTransitionGraph,
+    initial_state: Sequence[StateWithSpins],
+    final_state: Sequence[StateWithSpins],
+) -> Generator[Dict[int, StateWithSpins], None, None]:
```

Comment:
Does this function does not need the `StateWithSpins` right? In other words a particle name or pid would suffice. This connects to the question below

Reply:
Oh oops, this commit actually belongs to #245

Reply:
I moved the commit to #245. But still, this `generate_outer_edge_permutations` uses the output of `_safe_set_spin_projections`, which is why the properties are `StateWithSpins` at this stage. Can also make it more abstract by using `Any`, because internally this function does not need `StateWithSpins`.

Reply:
making it more abstract sounds good to me